### DISCUSSION TITLE
Prevent destruction of entire pages

### DISCUSSION THREAD

**Discussion by clonefunnels** [GrapesJS/grapesjs#4040] [Ideas]
I've always wondered if there is a reason that if there is one small error in the code, it will destroy the entire page.  If you save it, you're done.

Is there a way you could prevent this?

![2022-01-01_04h21_33](https://user-images.githubusercontent.com/95948977/147847712-5d92eb55-dd2c-490a-a2a1-2b140b03eccd.png)

**Comment by artf** [comment]
You can implement your custom importer with your checks before passing it to the editor.

**Comment by clonefunnels** [comment]
I'm doing about 10,000 string replaces to fix things that break it.  But it
would be nice if it just didn't break.
